app/src/main/java/com/example/proptit_protify/song_detail/SongDetailFragment.kt

```diff
@@ -1,60 +1,74 @@
-        }
-    }
-
+    private var _binding: FragmentSongDetailBinding? = null
+    private val binding get() = _binding!!
+    private val songDetailViewModel: SongDetailViewModel by viewModels()
+    private var selectedTrack: Track? = null
+    private var idTrack: Long = 0
+    private lateinit var exoPlayer: ExoPlayer
+    private lateinit var mediaItem: MediaItem
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
+        inflater: LayoutInflater,
+        container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_song_detail, container, false)
+    ): View {
+        _binding = FragmentSongDetailBinding.inflate(inflater, container, false)
+        return binding.root
     }
-
-    companion object {
-        /**
-         * Use this factory method to create a new instance of
-         * this fragment using the provided parameters.
-         *
-         * @param param1 Parameter 1.
-         * @param param2 Parameter 2.
-         * @return A new instance of fragment MusicDetailFragment.
-         */
-        // TODO: Rename and change types and number of parameters
-        @JvmStatic
-        fun newInstance(param1: String, param2: String) =
-            SongDetailFragment().apply {
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
-                }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.downButton.setOnClickListener {
+            findNavController().popBackStack()
+        }
+        idTrack = arguments?.getLong("idTrack") ?: 6461440
+        songDetailViewModel.findTrack(idTrack)
+        songDetailViewModel.track.observe(viewLifecycleOwner){track->
+            selectedTrack = track
+            binding.name.text = selectedTrack?.title
+            binding.artist.text = selectedTrack?.artist?.name
+            Glide.with(this)
+                .load(selectedTrack?.album?.cover)
+                .error(R.drawable.ic_pause)
+                .placeholder(R.drawable.ic_download)
+                .into(binding.avatar)
+            /*
+            val intent = Intent(context, MediaPlaybackService::class.java).apply {
+                putExtra("songUri", selectedTrack?.preview)
             }
+            context?.startService(intent)
+             */
+            exoPlayer = ExoPlayer.Builder(requireContext()).build()
+            binding.playerControl.player = exoPlayer
+            mediaItem = MediaItem.fromUri(selectedTrack?.preview?.toUri() ?:
+            "https://cdns-preview-1.dzcdn.net/stream/c-1ed50e5b3118c99be858fc305609e62a-15.mp3".toUri())
+            exoPlayer.setMediaItem(mediaItem)
+            exoPlayer.prepare()
+            exoPlayer.play()
```

Comment:
this fragment and the service has their own exoplayer. so they could play 2 song seperately, which is an error.